libr/core/core.c

```diff
@@ -122,6 +122,44 @@ R_API ut64 r_core_get_asmqjmps(RCore *core, const char *str) {
 	return UT64_MAX;
 }
 
+/**
+ * Takes addr and returns already saved shortcut or a new one
+ * The returned buffer needs to be freed
+ */
+R_API char* r_core_add_asmqjmp(RCore *core, ut64 addr) {
+	int i, found = 0;
+	if (!core->asmqjmps) return NULL;
+	if (core->is_asmqjmps_letter) {
+		if (core->asmqjmps_count >= R_CORE_ASMQJMPS_MAX_LETTERS) {
+			return NULL;
+		}
+
+		if (core->asmqjmps_count >= core->asmqjmps_size - 2) {
+			core->asmqjmps_size *= 2;
```

Comment:
I think it's better to update the size after knowing the reallocation went ok, because otherwise if it fails you would think the size is the double.